Makefile

```
@@ -67,25 +70,48 @@ test-all: go-test python-test ## Run all tests
+
+.PHONY: test-release-matrix
+test-release-matrix: ## Run release readiness regression matrix
+	@mkdir -p artifacts
+	cd python && PYTHONHASHSEED=0 pytest tests/regression/ -v -m "security or idempotency or slow" --no-cov || [ $$? -eq 5 ]
+	bash scripts/perf-budget-benchmark.sh artifacts/perf-budget-report.txt
+
+.PHONY: test-ci-gate
+test-ci-gate: ci-precheck go-test python-test lint-all security-scan ## Deterministic CI gate
+
+.PHONY: test-matrix
+test-matrix: ## Run full test matrix (CI gate + staging + release matrix)
```

checkmake (0.3.2)

[warning] Target body for "test-matrix" exceeds allowed length of 5 lines (7).

(maxbodylength)